rio_tiler/io/xarray.py

```diff
@@ -223,6 +223,10 @@ def tile(
         arr = ds.to_masked_array()
         arr.mask |= arr.data == ds.rio.nodata
 
+        output_bounds = ds.rio._unordered_bounds()
+        if output_bounds[1] > output_bounds[3] and ds.rio.transform().e > 0:
```

Comment:
We check if the `unordered bounds` are in form of `xmin, ymax, xmax, ymin` and if the transform's y resolution is positive.

Those 2 clues tells us the image if flipped 

I don't know enough about NetCDF/rioxarray to tell if this will be enough